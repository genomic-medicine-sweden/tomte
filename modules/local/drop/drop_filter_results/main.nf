process DROP_FILTER_RESULTS {
    tag "DROP_FILTER_RESULTS"
    label 'process_low'

    // Exit if running this module with -profile conda / -profile mamba
    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
        exit 1, "Local DROP module does not support Conda. Please use Docker / Singularity / Podman instead."
    }

    container "docker.io/clinicalgenomics/drop:1.4.0"

    input:
    val(case_info)
    path gene_panel_clinical_filter
    path out_drop_ae_rds_in
    path out_drop_gene_name_in
    path out_drop_as_tsv_in

    output:
    path('*outrider_top_hits_research.tsv') , optional: true, emit: ae_out_research
    path('*outrider_top_hits_clinical.tsv') , optional: true, emit: ae_out_clinical
    path('*fraser_top_hits_research.tsv')   , optional: true, emit: as_out_research
    path('*fraser_top_hits_clinical.tsv')   , optional: true, emit: as_out_clinical
    path "versions.yml"                                      , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def ids = "${case_info.probands}".replace("[","").replace("]","").replace(",","")
    def case_id = "${case_info.id}".replace("[","").replace("]","").replace(",","")
    def gene_panel_filter = gene_panel_clinical_filter ? "--gene_panel ${gene_panel_clinical_filter}" : ''
    def drop_ae_rds = out_drop_ae_rds_in ? "--drop_ae_rds ${out_drop_ae_rds_in}" : ''
    def out_drop_gene_name = out_drop_gene_name_in ? "--out_drop_gene_name ${out_drop_gene_name_in}" : ''
    def out_drop_as_tsv = out_drop_as_tsv_in ? "--out_drop_as_tsv ${out_drop_as_tsv_in}" : ''

    """
    drop_filter_results.py \\
        --samples $ids \\
        $gene_panel_filter \\
        $drop_ae_rds \\
        $out_drop_gene_name \\
        $out_drop_as_tsv \\
        --case $case_id \\

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        drop_filter_results: \$(drop_filter_results.py --version)
    END_VERSIONS
    """

    stub:
    """
    touch outrider_top_hits_research.tsv
    touch outrider_top_hits_clinical.tsv
    touch fraser_top_hits_research.tsv
    touch fraser_top_hits_clinical.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        drop_filter_results: \$(drop_filter_results.py --version)
    END_VERSIONS
    """
}
