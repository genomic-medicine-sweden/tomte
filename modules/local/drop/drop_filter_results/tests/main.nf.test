nextflow_process {

    name "Test Process DROP_FILTER_RESULTS"
    script "../main.nf"
    process "DROP_FILTER_RESULTS"
    tag "modules"
    tag "modules_local"
    tag "drop"
    tag "drop_filter_results"

    setup {
        run("GUNZIP", alias: "GUNZIP_FASTA") {
            script "../../../../../modules/nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'fasta' ],
                    file("${projectDir}/test_data/grch37_chr21.fa.gz", checkIfExists: true)
                ])
                """
            }
        }

        run("SAMTOOLS_FAIDX") {
            script "../../../../../modules/nf-core/samtools/faidx/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip
                input[1] = [[],[]]
                input[2] = []
                """
            }
        }

        run("GUNZIP", alias: "GUNZIP_GTF") {
            script "../../../../../modules/nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'gtf' ],
                    file("${projectDir}/test_data/grch37_chr21.gtf.gz", checkIfExists: true)
                ])
                """
            }
        }

        run("DROP_SAMPLE_ANNOT") {
            script "../../../../../modules/local/drop/drop_sample_annot/main.nf"
            process {
                """
                input[0] = Channel.of([
                            ['ACC5963A1'], [false], ['reverse'], ['NA'],
                            file("${projectDir}/test_data/ACC5963A1.bam", checkIfExists: true),
                            file("${projectDir}/test_data/ACC5963A1.bam.bai", checkIfExists: true)
                        ])
                input[1] = channel.fromPath("${projectDir}/test_data/drop_data/geneCounts.tsv.gz", checkIfExists: true)
                input[2] = channel.fromPath("${projectDir}/test_data/drop_data/sampleAnnotation.tsv", checkIfExists: true)
                input[3] = 'outrider'
                input[4] = 'fraser'
                """
            }
        }

        run("DROP_CONFIG_RUN_AS") {
            script "../../../../../modules/local/drop/drop_config_runAS/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip.join(SAMTOOLS_FAIDX.out.fai).collect()
                input[1] = GUNZIP_GTF.out.gunzip.collect()
                input[2] = DROP_SAMPLE_ANNOT.out.drop_annot
                input[3] = Channel.of([
                    file("${projectDir}/test_data/ACC5963A1.bam", checkIfExists: true),
                    file("${projectDir}/test_data/ACC5963A1.bam.bai", checkIfExists: true)
                    ])
                input[4] = channel.fromPath("${projectDir}/test_data/drop_data/geneCounts.tsv.gz", checkIfExists: true)
                input[5] = channel.fromPath("${projectDir}/test_data/drop_data", checkIfExists: true)
                input[6] = 'GRCh37'
                input[7] = 'fraser'
                input[8] = 'outrider'
                input[9] = 1
                input[10] = true
                """
            }
        }

        run("DROP_CONFIG_RUN_AE") {
            script "../../../../../modules/local/drop/drop_config_runAE/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip.join(SAMTOOLS_FAIDX.out.fai).collect()
                input[1] = GUNZIP_GTF.out.gunzip.collect()
                input[2] = DROP_SAMPLE_ANNOT.out.drop_annot
                input[3] = Channel.of([
                        file("${projectDir}/test_data/ACC5963A1.bam", checkIfExists: true),
                        file("${projectDir}/test_data/ACC5963A1.bam.bai", checkIfExists: true)
                    ])
                input[4] = channel.fromPath("${projectDir}/test_data/drop_data/geneCounts.tsv.gz", checkIfExists: true)
                input[5] = channel.fromPath("${projectDir}/test_data/drop_data", checkIfExists: true)
                input[6] = 'GRCh37'
                input[7] = 'outrider'
                input[8] = 'fraser'
                input[9] = 1
                input[10] = 2.5
                input[11] = true
                """
            }
        }

    }

    test("With AS and AE") {

        when {
            process {
                """
                input[0] = Channel.of([probands:['ACC5963A1'], id:'finequagga']).collect()
                input[1] = channel.fromPath("${projectDir}/test_data/drop_data/mock_gene_panel.tsv", checkIfExists: true).collect()
                input[2] = DROP_CONFIG_RUN_AE.out.drop_ae_rds
                input[3] = DROP_CONFIG_RUN_AS.out.drop_gene_name
                input[4] = DROP_CONFIG_RUN_AS.out.drop_as_tsv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.drop_as_out_clinical.collect{ file(it).name },
                    process.out.drop_as_out_research.collect{ file(it).name },
                    process.out.drop_ae_out_clinical.collect{ file(it).name },
                    process.out.drop_ae_out_research.collect{ file(it).name },
                    process.out.versions,
                ).match()}
            )
        }

    }

    test("With only AE") {

        when {
            process {
                """
                input[0] = Channel.of([probands:['ACC5963A1'], id:'finequagga']).collect()
                input[1] = channel.fromPath("${projectDir}/test_data/drop_data/mock_gene_panel.tsv", checkIfExists: true).collect()
                input[2] = DROP_CONFIG_RUN_AE.out.drop_ae_rds
                input[3] = DROP_CONFIG_RUN_AE.out.drop_gene_name
                input[4] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.drop_ae_out_clinical.collect{ file(it).name },
                    process.out.drop_ae_out_research.collect{ file(it).name },
                    process.out.versions,
                ).match()}
            )
        }

    }

    test("With only AS") {

        when {
            process {
                """
                input[0] = Channel.of([probands:['ACC5963A1'], id:'finequagga']).collect()
                input[1] = channel.fromPath("${projectDir}/test_data/drop_data/mock_gene_panel.tsv", checkIfExists: true).collect()
                input[2] = []
                input[3] = DROP_CONFIG_RUN_AS.out.drop_gene_name
                input[4] = DROP_CONFIG_RUN_AS.out.drop_as_tsv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.drop_as_out_clinical.collect{ file(it).name },
                    process.out.drop_as_out_research.collect{ file(it).name },
                    process.out.versions,
                ).match()}
            )
        }

    }

}
