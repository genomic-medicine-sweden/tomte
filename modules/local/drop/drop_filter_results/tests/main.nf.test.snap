{
    "With only AS": {
        "content": [
            [
                "finequagga_fraser_top_hits_clinical.tsv"
            ],
            [
                "finequagga_fraser_top_hits_research.tsv"
            ],
            [
                "versions.yml:md5,1c377afea6896740ff3d37642c9c1608"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.5"
        },
        "timestamp": "2025-09-10T16:49:17.300088668"
    },
    "With only AE": {
        "content": [
            [
                "finequagga_outrider_top_hits_clinical.tsv"
            ],
            [
                "finequagga_outrider_top_hits_research.tsv"
            ],
            [
                "versions.yml:md5,1c377afea6896740ff3d37642c9c1608"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.5"
        },
        "timestamp": "2025-09-10T16:41:13.393007585"
    },
    "With AS, AE, and MAE": {
        "content": [
            [
                "finequagga_fraser_top_hits_clinical.tsv"
            ],
            [
                "finequagga_fraser_top_hits_research.tsv"
            ],
            [
                "finequagga_outrider_top_hits_clinical.tsv"
            ],
            [
                "finequagga_outrider_top_hits_research.tsv"
            ],
            [
                "finequagga_mae_top_hits_clinical.tsv"
            ],
            [
                "finequagga_mae_top_hits_research.tsv"
            ],
            [
                "versions.yml:md5,1c377afea6896740ff3d37642c9c1608"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.5"
        },
        "timestamp": "2025-09-10T19:05:25.487309998"
    }
}